pool/src/lib.rs

```diff
@@ -709,23 +749,64 @@ impl LendingPool {
         assert_with_error!(env, value > 0, Error::MustBePositive);
     }
 
+    /// Check that balance + deposit + debt * ar_lender <= reserve.configuration.liq_cap
+    fn require_liq_cap_not_exceeded(
+        env: &Env,
+        reserve: &ReserveData,
+        balance: i128,
+        deposit_amount: i128,
+    ) -> Result<(), Error> {
+        let balance_after_deposit = {
+            let debt_token = DebtTokenClient::new(env, &reserve.debt_token_address);
+            let debt_supply = debt_token.total_supply();
+
+            FixedI128::from_inner(reserve.lender_accrued_rate)
+                .mul_int(debt_supply)
+                .ok_or(Error::MathOverflowError)?
+                .checked_add(deposit_amount)
+                .ok_or(Error::MathOverflowError)?
+                .checked_add(balance)
+                .ok_or(Error::MathOverflowError)?
+        };
+
+        assert_with_error!(
+            env,
+            balance_after_deposit <= reserve.configuration.liq_cap,
+            Error::LiqCapExceeded
+        );
+
+        Ok(())
+    }
+
     fn do_deposit(
         env: &Env,
         who: &Address,
         asset: &Address,
         amount: i128,
         reserve: &ReserveData,
     ) -> Result<bool, Error> {
-        let collat_coeff = Self::get_collateral_coeff(env, reserve)?;
+        if amount == 0 {
```

Comment:
Let's remove the `remaining_amount` check since you do it in the `do_deposit` method.
```
let is_first_deposit = if remaining_amount > 0 {
            Self::do_deposit(&env, &who, &asset, remaining_amount, &reserve)?
        } else {
            false
        };
```